cpp/dolfinx/mesh/utils.cpp

```diff
@@ -60,48 +60,44 @@ std::vector<double> mesh::h(const Mesh& mesh,
                             const xtl::span<const std::int32_t>& entities,
                             int dim)
 {
-  if (dim != mesh.topology().dim())
-    throw std::runtime_error("Cell size when dim ne tdim  requires updating.");
 
-  if (mesh.topology().cell_type() == CellType::prism and dim == 2)
-    throw std::runtime_error("More work needed for prism cell");
-
-  // Get number of cell vertices
-  const CellType type = cell_entity_type(mesh.topology().cell_type(), dim, 0);
-  const int num_vertices = num_cell_vertices(type);
+  // Get entity vertices in geometry dofs
+  xt::xtensor<std::int32_t, 2> geometry_vertices
+      = entities_to_geometry(mesh, dim, entities, false);
+  const std::size_t num_vertices_per_entity = geometry_vertices.shape(1);
 
   // Get geometry dofmap and dofs
-  const Geometry& geometry = mesh.geometry();
-  const graph::AdjacencyList<std::int32_t>& x_dofs = geometry.dofmap();
-  auto geom_dofs
-      = xt::adapt(geometry.x().data(), geometry.x().size(), xt::no_ownership(),
-                  std::vector({geometry.x().size() / 3, std::size_t(3)}));
-  std::vector<double> h_cells(entities.size(), 0);
-  assert(num_vertices <= 8);
-  xt::xtensor_fixed<double, xt::xshape<8, 3>> points;
-  for (std::size_t e = 0; e < entities.size(); ++e)
+  const mesh::Geometry& geometry = mesh.geometry();
+  const xtl::span<const double> x_g = geometry.x();
+
+  // Lambda function to compute norm
+  auto l2_norm = [](auto p0, auto p1)
```

Comment:
The `auto` make things a bit unclear. The return type is always `double`, so it would be clearer if it was explicit.

I'm not sure what `p0` and `p1` are. There are index into, but from where the function is called it looks like an xtensor iterator.